### DISCUSSION TITLE
Allow polling only in foreground

### DISCUSSION THREAD

**Discussion by benborgers** [livewire/livewire#1622] [4. General]
I'm running an app that relies heavily on polling to create a "real-time" UI that's the same for all users. 

The app gets used during live college classes, but I've noticed thousands of requests coming in for some users hours after the class ends. I suspect that these people have the tab open in the background, given that the requests are coming in ~5% as often as they usually would if the tab was foregrounded. 

Is there any interest in me opening a PR for a `wire:poll.foreground` (name suggestions welcome!) modifier that **only** polls when the tab is in the foreground? I think that this could prevent a lot of unnecessary server requests in some situations, and the page would be fully updated a few seconds after the user returns to that tab.

**Comment by ju5t** [comment]
I think this is a cool idea. We don't really need it, but I would use it.
